### ISSUE TITLE
wrong URL after language switch when using custom language

### ISSUE THREAD

**Issue by DiegoAceve** [qtranslate/qtranslate-xt#1035]
Hi,

I have added a new language (Papiamento) and used 3 letter code "pap" when creating it.  After adding the language switcher menu and choosing this language links are now: https://website.com/papp/page/  instead of: https://website.com/pap/page/

All other languages are working fine, appreciate if someone knows how to fix this.

**Comment by herrvigg** [comment]
I've tried on a test website and it works as expected, the 3-letter language code is handled correctly in the menu.
I don't have any URL with `page` literally but it works with all the pages I have.
Do you have other plugins enabled? qtranslate-slug? Also check if you have some specific URL rewrites.

**Comment by herrvigg** [comment]
Oh wait, I see the problem with one menu... just one. How strange. I'll check this.

**Comment by MK-RD** [comment]
QTX 3.10.0

[ qtranslate_core.php ][ qtranxf_url_del_language() ]

```
...

#! $urlinfo['wp-path'] = substr($urlinfo['wp-path'], 3);
$urlinfo['wp-path'] = substr($urlinfo['wp-path'], strlen($match[1]) + 1);

...
```

When the Solution is the Problem: Cleanup Time!

**Comment by herrvigg** [comment]
@MK-RD good catch!
@DiegoAceve Fixed in master.

This didn't only impact the language switcher menu but possibly all menus with a 3-letters language code.

Still something weird as some menus are sent to `qtranxf_convertURL` with the language prefix and others not, while they all come from the same kind of pages. That explains why some menus were still correct, but I don't know the reason yet. Anyway that's a different question and the fix here should solve the current issues.

**Comment by herrvigg** [comment]
Released in 3.11.0.
